File: LeetCode/2_week_study_plan_to_tackle_ds/data_structure_1/java/ContainsDuplicate.java
import java.util.ArrayList;

class ContainsDuplicate{
    public static boolean contains(int[] nums) {
        Boolean contains = false;
        ArrayList<Integer> arr = new ArrayList<Integer>();

        for(int value : nums){
            arr.add(value);
        }

        for(Integer value : arr){
            if(arr.indexOf(value) != arr.lastIndexOf(value)){
                contains = true;
            }
        }

        return contains;
    }
    public static void main(String[] args){
        int[] nums = {1,2,3};
        System.out.println(contains(nums));
    }
}

File: LeetCode/2_week_study_plan_to_tackle_ds/data_structure_1/javascript/containsDuplicate.js
/* Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

 

Example 1:

Input: nums = [1,2,3,1]
Output: true

Example 2:

Input: nums = [1,2,3,4]
Output: false

Example 3:

Input: nums = [1,1,1,3,3,4,3,2,4,2]
Output: true
 */

/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
    let contains = false
    nums.forEach(value=>{
        if(nums.lastIndexOf(value) !== nums.indexOf(value)){
            contains = true
        }
    })

    return contains
};

console.log(containsDuplicate([1,2,3,1]))